lib/chef/knife/core/cookbook_scm_repo.rb

```diff
@@ -130,15 +130,23 @@ def get_current_branch()
       private
 
       def git_repo?(directory)
-        if File.directory?(File.join(directory, '.git'))
+        possible_git_directory = File.join(directory, '.git')
+        if File.directory?(possible_git_directory)
+          return true
+        elsif File.exists?(possible_git_directory) and is_git_dir_redirect?(possible_git_directory)
```

Comment:
While I don't think it matters in this case, you should be careful using `and` and `or`:
http://devblog.avdi.org/2014/08/26/how-to-use-rubys-english-andor-operators-without-going-nuts/